consensus/babe/verify.go

```diff
@@ -18,3 +20,76 @@ func (b *Session) verifySlotWinner(slot uint64, header *BabeHeader) (bool, error
 
 	return pub.VrfVerify(vrfInput, header.VrfOutput[:], header.VrfProof[:])
 }
+
+func (b *Session) verifyAuthorshipRight(slot uint64, header *types.BlockHeader) (bool, error) {
+	// header should have 2 digest items (possibly more in the future)
+	// first item should be pre-digest, second should be seal
+	if len(header.Digest) < 2 {
+		return false, fmt.Errorf("block header is missing digest items")
+	}
+
+	// check for valid seal by verifying signature
+	preDigestBytes := header.Digest[0]
+	sealBytes := header.Digest[len(header.Digest)-1]
```

Comment:
Is there a reason this is not `sealBytes := header.Digest[1]`?

Reply:
the seal is always the last digest item, not necessarily the second. there might be other digests in between digest[0] and digest[len-1].